tests/basic/pure_test.go

```diff
@@ -4238,3 +4238,101 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
+		1. Deploy an app which uses FADA volume( uses csi provisioner and a pure_block backend) and an app which uses base volume(uses portworx-volume as provisioner)
+		2. Parallely resize the pvc of FADA deployed app and also resize the base app pvc
+		3. Validate the volume resize
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("ValidateVolumeResizeInParallel", "Px Volume Resize in parallel to FADA/FBDA Volume Resize ( PVC Resize )", nil, 0)
+	})
+	var contexts []*scheduler.Context
+	itLog := "Px Volume Resize in parallel to FADA/FBDA Volume Resize ( PVC Resize )"
+	It(itLog, func() {
+		log.InfoD(itLog)
+
+		var wg sync.WaitGroup
+		var resizedVols []*volume.Volume
+		appList := Inst().AppList
+		defer func() {
+			Inst().AppList = appList
+		}()
+		stepLog := "Deploy applications"
+		Step(stepLog, func() {
+			Inst().AppList = []string{"fio-cloudsnap"}
+			appNamespace := fmt.Sprintf("volumeresizeparallel-%s", Inst().InstanceID)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplicationsOnNamespace(appNamespace, "volumeresizeparallel")...)
+			}
+		})
+		stepLog = "Deploy a app which uses FADA volume and validate"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			context, err := Inst().S.Schedule("volumeresizeparallelfada", scheduler.ScheduleOptions{
+				AppKeys:            []string{"fio-fa-davol"},
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", "volumeresizeparallelfada")
+			contexts = append(contexts, context...)
+
+		})
+		ValidateApplications(contexts)
+		resizeVolumes := func(ctx *scheduler.Context) {
+			var pvcs []*v1.PersistentVolumeClaim
+			appVolumes, err := Inst().S.GetVolumes(ctx)
+			log.FailOnError(err, "Failed to get volumes from context")
+			log.InfoD(fmt.Sprintf("increase volume size %s on app %s's volumes: %v",
+				Inst().V.String(), ctx.App.Key, appVolumes))
+			for _, eachVol := range appVolumes {
+				pvc, err := GetPVCObjFromVol(eachVol)
+				log.FailOnError(err, "Failed to get PVC Details from Volume [%v]", eachVol.Name)
+				pvcs = append(pvcs, pvc)
+			}
+			for _, pvc := range pvcs {
+				pvcSize := pvc.Spec.Resources.Requests.Storage().String()
+				pvcSize = strings.TrimSuffix(pvcSize, "Gi")
+				pvcSizeInt, err := strconv.Atoi(pvcSize)
+				log.InfoD("increasing pvc [%s/%s]  size to %v %v", pvc.Namespace, pvc.Name, 2*pvcSizeInt, pvc.UID)
+				resizedVol, err := Inst().S.ResizePVC(ctx, pvc, uint64(pvcSizeInt))
+				log.FailOnError(err, "pvc resize failed pvc:%v", pvc.UID)
+				log.InfoD("Vol uid %v", resizedVol.ID)
+				resizedVols = append(resizedVols, resizedVol)
+			}
+		}
+		stepLog = "Do parallel resize of base apps volume and FADA volume"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				defer GinkgoRecover()
+				for _, ctx := range contexts {
+					resizeVolumes(ctx)
```

Comment:
this will be running a sequentially as your for loop is inside go routine. if you want to run in parallel start go routine inside for loop